### PR TITLE
Feat: Add access to coin flow in the generosity challenge (kids-1297)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
  - Introduced a floating action button in the Generosity Challenge Overview page to start specific flows and navigate to related pages.
  - Added a new coin icon option for use in the application.
  - Implemented support for in-app generosity coin flow in various features.
  
- **UI Enhancements**
  - Refactored the layout of the Generosity Challenge Overview page to improve user experience.

- **Bug Fixes**
  - Improved handling of scanned NFC events to ensure correct navigation based on flow type.

- **Event Logging**
  - Added new event types for better tracking of user interactions related to generosity challenges.

### CODE REVIEW

File: lib/features/children/generosity_challenge/pages/generosity_challenge_overview.dart [hunk 1]
```diff
@@ -169,35 +172,50 @@ class _GenerosityChallengeOverviewState
...
-                      Text(UnlockDayTimeDifference.minutes.name),
-                    ],
                   ),
+                const Spacer(),
               ],
             ),
           ],
         ),
       ),
+      floatingActionButton: GivtElevatedButton(
```
Comment (MaikelStuivenberg) [inline lib/features/children/generosity_challenge/pages/generosity_challenge_overview.dart:207]: I don't think it should be an floatingactionbutton, but just a button at the bottom of the page.
floatingactionbutton is only if it needs to be on top of other things

**Review by TammiLion — APPROVED** [review]

**Comment by TammiLion** [comment]
Just take a look at maikels comment before merge.

**Comment by Daniela510** [comment]
> Don't forget to look at code rabbits comments ;) You have a duplicate import (according to the tabbit). And it might have good comments around refactor

i looked, its not duplicate import, we have the import of legacy givt4kids utils on the page since its a g4k legacy page, and importing family/utils to import the FamilyAppTheme.

**Comment by Daniela510** [comment]
The other rabbit comments have to do with refactoring the NFCScanPage, wich is outside scope for the specif ticket this PR is meant for.

**Pull request merged**